packages/grafana-i18n/src/i18n.tsx

```diff
@@ -158,11 +174,40 @@ export function t(id: string, defaultMessage: string, values?: Record<string, un
   return tFunc(id, defaultMessage, values);
 }
 
+function useInit(entity: string) {
+  const [isInitialized, setIsInitialized] = useState(false);
+
+  useEffect(() => {
+    async function init() {
+      if (!tFunc || !transComponent) {
+        logWarning(entity);
+        await initPluginTranslations(entity);
+      }
+      setIsInitialized(true);
+    }
+    init();
+  }, [entity]);
+
+  return isInitialized;
+}
+
 export function useTranslate() {
+  const isInitialized = useInit('useTranslate()');
```

Comment:
I'm a bit confused - how come we call this with `useTranslate()`, as that then gets passed to `initPluginTranslations`, which ends up with us having...

```tsx
transComponent = (props: TransProps) => <I18NextTrans shouldUnescape ns={'useTranslate()'} {...props} />;
```

unless I'm not following the flow correctly :grin:

Reply:
I completely changed the approach because using a hook caused a lot of failing unit tests that depend on components not being re-rendered. Added a self executing function instead and sane defaults